File: templates/cpp/@DAY@_test.cc
#include "@DAY@.h"

#include <gtest/gtest.h>

#include <tuple>
#include <vector>

auto checkTC =
    [](const auto& tc, auto checkfun) {
      auto& in = std::get<0>(tc);
      auto& want = std::get<1>(tc);
      auto got = checkfun(in);
      EXPECT_EQ(want, got);
    }

TEST(@DAY@, @DAY@PartOne) {
  using TestCase = std::tuple<std::vector<std::string>, int>;
  TestCase tc{{""}, 0};
  checkTC(tc, @DAY@::puzzle1);
}

TEST(@DAY@, @DAY@PartTwo) {
  using TestCase = std::tuple<std::vector<std::string>, int>;
  TestCase tc{{""}, 0};
  checkTC(tc, @DAY@::puzzle2);
}

File: templates/go/@DAY@_test.go
package main

import (
	"reflect"
	"testing"
)

func Test_@DAY@_puzzle1(t *testing.T) {
	tests := []struct {
		name  string
		input []string
		want  int
	}{
		{name: "Example1", input: []string{"1"}, want: 1},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := puzzle1(tt.input); !reflect.DeepEqual(got, tt.want) {
				t.Errorf("Test_@DAY@_puzzle1() = %v, want %v", got, tt.want)
			}
		})
	}
}

func Test_@DAY@_puzzle2(t *testing.T) {
	tests := []struct {
		name  string
		input []string
		want  int
	}{
		{name: "Example1", input: []string{"1"}, want: 1},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := puzzle2(tt.input); !reflect.DeepEqual(got, tt.want) {
				t.Errorf("Test_@DAY@_puzzle2() = %v, want %v", got, tt.want)
			}
		})
	}
}
